flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -296,6 +296,11 @@ private MemorySegment requestMemorySegment(int targetChannel) {
 		synchronized (availableMemorySegments) {
 			returnExcessMemorySegments();
 
+			// target channel over quota; do not return a segment
```

Comment:
Shouldn't this commit change some tests?

Reply:
Yes it changes `LocalBufferPoolTest#testMaxBuffersPerChannelAndAvailability`. Other test played nicely (only took buffer they should have taken).

Reply:
Pulled out the test changes into this commit.